### PR TITLE
fix(redteam): Do not fail crescendo if the provider sends the wrong response

### PR DESCRIPTION
**Pull request by sklein12**
We were overzealous in failing crescendo attacks before. If we don't get an attack prompt we'll just skip a turn and keep going.

Chore: Added [Crescendo] prefix to all log statements in the provider

### CODE REVIEW

File: src/redteam/providers/crescendo/index.ts [hunk 1]
```diff
@@ -391,22 +401,22 @@ class CrescendoProvider implements ApiProvider {
 
     for (const key of expectedKeys) {
       if (!(key in parsedOutput)) {
-        logger.error(`Missing key in response: ${key}`);
-        throw new Error(`Expected key '${key}' not found in JSON response: ${response.output}`);
```
Comment (sklein12) [inline src/redteam/providers/crescendo/index.ts:395]: This is the actual change

File: src/redteam/providers/crescendo/index.ts [hunk 2]
```diff
@@ -391,22 +401,22 @@ class CrescendoProvider implements ApiProvider {
...
       rationaleBehindJailbreak: ${parsedOutput.rationaleBehindJailbreak}
       lastResponseSummary: ${parsedOutput.lastResponseSummary}
     `);
 
     if (Object.keys(parsedOutput).length !== expectedKeys.length) {
-      logger.error(`Unexpected keys in response: ${Object.keys(parsedOutput).join(', ')}`);
-      throw new Error(`Unexpected keys found in JSON response: ${response.output}`);
+      logger.debug(
+        `[Crescendo] Unexpected keys in response: ${Object.keys(parsedOutput).join(', ')}`,
+      );
```
Comment (sklein12) [inline src/redteam/providers/crescendo/index.ts:419]: This is the second actual change

File: src/redteam/providers/crescendo/index.ts [hunk 3]
```diff
@@ -209,7 +211,13 @@ class CrescendoProvider implements ApiProvider {
         totalTokenUsage.numRequests += attackTokenUsage.numRequests ?? 1;
         totalTokenUsage.cached += attackTokenUsage.cached || 0;
       }
-      logger.debug(`Generated attack prompt: ${attackPrompt}`);
+
+      if (!attackPrompt) {
+        logger.error('[Crescendo] failed to generate a question. Will skip turn and try again');
+        continue;
+      }
+
```
Comment (sklein12) [inline src/redteam/providers/crescendo/index.ts:219]: This is the third and final actual change

**Review by typpo — APPROVED** [review]

**Review by mldangelo — APPROVED** [review]
Looks good! Thank you!

**Pull request merged**